### ISSUE TITLE
Pipeline breaking task does not detect scan

### ISSUE THREAD

**Issue by JuanFelipeRamirezBolivar** [simondel/sonar-buildbreaker-vsts#35]
Pipelines do not break when using the latest version of SonarQube tasks (5*) as the pipeline breaking task does not detect any analysis.

![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/78444126/6d2592a2-a0af-4c9c-a483-88e90b944e6c)

![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/78444126/324fb215-dec0-48cd-a9b5-a0fce0c37ee6)

**Comment by devopslikeme** [comment]
[This response works for me.](https://github.com/simondel/sonar-buildbreaker-vsts/issues/32#issuecomment-1593250792)

**Comment by JuanFelipeRamirezBolivar** [comment]
> [Esta respuesta funciona para mí.](https://github.com/simondel/sonar-buildbreaker-vsts/issues/32#issuecomment-1593250792)

Thanks. We’ll be testing it, for now it also works with version 4* of the task.

**Comment by Jan-Pleva** [comment]
We have same problem.

**Comment by yfrindel** [comment]
Hi. We have the same issue. The report shows various errors, but the "SonarQube build breaker" doesn't fail the build :
![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/22955005/0b7b7c7a-fe54-407d-85e1-4327498dfe68)
![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/22955005/8435caf6-a918-406b-bdd4-9f75967575d0)
![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/22955005/45f8ee22-6004-4f8b-9ee5-6571a68c1916)
![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/22955005/0eccb719-3484-4c80-ab96-f3788360dedc)
![image](https://github.com/simondel/sonar-buildbreaker-vsts/assets/22955005/0dc9cf46-fd0d-4925-86e2-cd2d0ee87687)
Do you think it will be fixed soon please?
Thanks

**Comment by KamilVolf** [comment]
Hi everyone,

Our workaround that works is to move the test results to the work directory. After this step, the quality breaker starts working.
example
  - task: CopyFiles@2
    displayName: 'Copy quality gate result to work directory'
    inputs:
       SourceFolder: '$(Agent.TempDirectory)/sonar'
       TargetFolder: '$(Common.TestResultsDirectory)'
    continueOnError: true

**Comment by ronogle** [comment]
We had the same problem.  However, I think that the behavior of the application is wrong.  If anything fails such as not finding the scan or any other issue that prevents this task from correctly determining if the scan failed or succeeded, then I think that this task needs to fail closed or secure instead of passing or allowing.  We use this task to break the build and we want to ensure that only builds that scan and pass are allowed.

Is this request different from this issue that I should open a new issue?

**Comment by asdkant-bf** [comment]
My workaround is a single bash task after the `SonarCloudPublish@3` task:
```yaml
bash: >-
  grep "Quality Gate passed" $(Build.ArtifactStagingDirectory)/.sqAnalysis/SonarQubeBuildSummary.md
```

Works without issues and you don't need to install anything 😉
I'm guessing that for windows you could do something very similar using powershell.

**Comment by htetbhs** [comment]
I have just tried the solution below from [KamilVolf](https://github.com/KamilVolf) and it works! Thank you...

> https://github.com/simondel/sonar-buildbreaker-vsts/issues/35#issuecomment-2173396347
